atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/keyvalue/TableMappingService.java

```diff
@@ -27,5 +27,5 @@
     TableReference getMappedTableName(TableReference tableRef) throws TableMappingNotFoundException;
     <T> Map<TableReference, T> mapToShortTableNames(Map<TableReference, T> tableMap)
             throws TableMappingNotFoundException;
-    Set<TableReference> mapToFullTableNames(Set<TableReference> tableNames);
+    Map<TableReference, TableReference> generateMapToFullTableNames(Set<TableReference> tableNames);
```

Comment:
There were no other uses of this method in the rest of this repo, but I'm not as familiar with the API guarantees of OSS Atlas.  Agreed that this is an API break and it's possible to make this change without breaking the API with something like the following in TableRemappingKeyValueService, but it feels more like a hack than a real solution.  Do you have a designated public API?  Is there a strong possibility that other products outside of Atlas have implemented the TableMappingService?:

```
 public Map<TableReference, byte[]> getMetadataForTables() {
        Map<TableReference, byte[]> unMappedTableRefToBytes = delegate().getMetadataForTables();
        //Pass in full set of unMappedTables to cache the tableMap and, more importantly, the unmappedTables.  This should
        //limit the number of times we need to re-read the tableMap from the DB for unmapped tables.
        tableMapper.mapToFullTableNames(unMappedTableRefToBytes.keySet());
        Map<TableReference, byte[]> mappedTableRefToBytes = Maps.newHashMapWithExpectedSize(unMappedTableRefToBytes.size());
        for (Entry<TableReference, byte[]> entry : unMappedTableRefToBytes.entrySet()) {
            mappedTableRefToBytes.put(
                    Iterables.getOnlyElement(tableMapper.mapToFullTableNames(ImmutableSet.of(entry.getKey()))),
                    entry.getValue());
        }
        return mappedTableRefToBytes;
    }